### PR TITLE
fixes symbolic link on py3 and windows

### LINKED ISSUE #2948: Provide compat that works with on Windows with Py 3.7.1
## Feature description
The script that creates the symbolic link doesn't work by default  when running `python -m spacy link en_core_web_sm en` -- normally if Dos is the default shell a call to `mklink` just works. However, in via this module, it does not and it is not a permission issues nor a system policy restriction as just running `mklink /d ...` from the same shell/command line works..

This fails regardless if under a virtualenv as well.

Instead a user is presented with the following error message.

```
(venv) C:\g\py\spacy> python -m spacy link en_core_web_sm en
C:\Program Files\Python37\lib\importlib\_bootstrap.py:219: RuntimeWarning: cymem.cymem.Pool size changed, may indicate binary incompatibility. Expected 48 from C header, got 64 from PyObject
  return f(*args, **kwds)
C:\Program Files\Python37\lib\importlib\_bootstrap.py:219: RuntimeWarning: cymem.cymem.Address size changed, may indicate binary incompatibility. Expected 24 from C header, got 40 from PyObject
  return f(*args, **kwds)

    Error: Couldn't link model to 'en'
    Creating a symlink in spacy/data failed. Make sure you have the required
    permissions and try re-running the command as admin, or use a
    virtualenv. You can still import the model as a module and call its
    load() method, or create the symlink manually.

    C:\g\py\spacy\venv\lib\site-packages\en_core_web_sm -->
    C:\g\py\spacy\venv\lib\site-packages\spacy\data\en

Traceback (most recent call last):
  File "C:\Program Files\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\g\py\spacy\venv\lib\site-packages\spacy\__main__.py", line 31, in <module>
    plac.call(commands[command], sys.argv[1:])
  File "C:\g\py\spacy\venv\lib\site-packages\plac_core.py", line 328, in call
    cmd, result = parser.consume(arglist)
  File "C:\g\py\spacy\venv\lib\site-packages\plac_core.py", line 207, in consume
    return cmd, self.func(*(args + varargs + extraopts), **kwargs)
  File "C:\g\py\spacy\venv\lib\site-packages\spacy\cli\link.py", line 48, in link
    symlink_to(link_path, model_path)
  File "C:\g\py\spacy\venv\lib\site-packages\spacy\compat.py", line 87, in symlink_to
    orig.symlink_to(dest)
  File "C:\Program Files\Python37\lib\pathlib.py", line 1320, in symlink_to
    self._accessor.symlink(target, self, target_is_directory)
OSError: symbolic link privilege not held
```

The following command works regardless and has no issue with permissions nor policy. The error message indicating symbolic link privilege not held is misleading.

```
cmd /c mklink /k c:\path\to\symlink c:\target\file
```

A suggestion is changing the respective lines in `venv\Lib\site-packages\spacy\compat.py (symlink_to)`

```
def symlink_to(orig, dest):
    if is_python2 and is_windows:
        import subprocess
        subprocess.call(['mklink', '/d', path2str(orig), path2str(dest)], shell=True)
    else:
        orig.symlink_to(dest)
```

to 

```
def symlink_to(orig, dest):
    if (is_python2 or is_python3) and is_windows:
        import subprocess
        subprocess.call(['mklink', '/d', path2str(orig), path2str(dest)], shell=True)
    else:
        orig.symlink_to(dest)
````

### PR DESCRIPTION
**Pull request by cicorias**
during setup of spacy using command
python -m spacy link en_core_web_sm en
closes #2948

## Description
fixes the issues address in #2948 - where the symbolic link isn't being created and the error message indicates permissions, while it is not permissions.

### Types of change
Bug fix

## Checklist

- [X] I have submitted the spaCy Contributor Agreement.
- [X] I ran the tests, and all new and existing tests passed.
- [X] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by cicorias** [comment]
# spaCy contributor agreement

This spaCy Contributor Agreement (**"SCA"**) is based on the
[Oracle Contributor Agreement](http://www.oracle.com/technetwork/oca-405177.pdf).
The SCA applies to any contribution that you make to any product or project
managed by us (the **"project"**), and sets out the intellectual property rights
you grant to us in the contributed materials. The term **"us"** shall mean
[ExplosionAI UG (haftungsbeschränkt)](https://explosion.ai/legal). The term
**"you"** shall mean the person or entity identified below.

If you agree to be bound by these terms, fill in the information requested
below and include the filled-in version with your first pull request, under the
folder [`.github/contributors/`](/.github/contributors/). The name of the file
should be your GitHub username, with the extension `.md`. For example, the user
example_user would create the file `.github/contributors/example_user.md`.

Read this agreement carefully before signing. These terms and conditions
constitute a binding legal agreement.

## Contributor Agreement

1. The term "contribution" or "contributed materials" means any source code,
object code, patch, tool, sample, graphic, specification, manual,
documentation, or any other material posted or submitted by you to the project.

2. With respect to any worldwide copyrights, or copyright applications and
registrations, in your contribution:

    * you hereby assign to us joint ownership, and to the extent that such
    assignment is or becomes invalid, ineffective or unenforceable, you hereby
    grant to us a perpetual, irrevocable, non-exclusive, worldwide, no-charge,
    royalty-free, unrestricted license to exercise all rights under those
    copyrights. This includes, at our option, the right to sublicense these same
    rights to third parties through multiple levels of sublicensees or other
    licensing arrangements;

    * you agree that each of us can do all things in relation to your
    contribution as if each of us were the sole owners, and if one of us makes
    a derivative work of your contribution, the one who makes the derivative
    work (or has it made will be the sole owner of that derivative work;

    * you agree that you will not assert any moral rights in your contribution
    against us, our licensees or transferees;

    * you agree that we may register a copyright in your contribution and
    exercise all ownership rights associated with it; and

    * you agree that neither of us has any duty to consult with, obtain the
    consent of, pay or render an accounting to the other for any use or
    distribution of your contribution.

3. With respect to any patents you own, or that you can license without payment
to any third party, you hereby grant to us a perpetual, irrevocable,
non-exclusive, worldwide, no-charge, royalty-free license to:

    * make, have made, use, sell, offer to sell, import, and otherwise transfer
    your contribution in whole or in part, alone or in combination with or
    included in any product, work or materials arising out of the project to
    which your contribution was submitted, and

    * at our option, to sublicense these same rights to third parties through
    multiple levels of sublicensees or other licensing arrangements.

4. Except as set out above, you keep all right, title, and interest in your
contribution. The rights that you grant to us under these terms are effective
on the date you first submitted a contribution to us, even if your submission
took place before the date you sign these terms.

5. You covenant, represent, warrant and agree that:

    * Each contribution that you submit is and shall be an original work of
    authorship and you can legally grant the rights set out in this SCA;

    * to the best of your knowledge, each contribution will not violate any
    third party's copyrights, trademarks, patents, or other intellectual
    property rights; and

    * each contribution shall be in compliance with U.S. export control laws and
    other applicable export and import laws. You agree to notify us if you
    become aware of any circumstance which would make any of the foregoing
    representations inaccurate in any respect. We may publicly disclose your
    participation in the project, including the fact that you have signed the SCA.

6. This SCA is governed by the laws of the State of California and applicable
U.S. Federal law. Any choice of law rules will not apply.

7. Please place an “x” on one of the applicable statement below. Please do NOT
mark both statements:

    * [ ] I am signing on behalf of myself as an individual and no other person
    or entity, including my employer, has or will have rights with respect to my
    contributions.

    * [ ] I am signing on behalf of my employer or a legal entity and I have the
    actual authority to contractually bind that entity.

## Contributor Details

| Field                          | Entry                |
|------------------------------- | -------------------- |
| Name                           |  Shawn Cicoria                     |
| Company name (if applicable)   |   Microsoft                   |
| Title or role (if applicable)  |   Principal Software Engineer                   |
| Date                           |   November  20, 2018                  |
| GitHub username                |     cicorias                 |
| Website (optional)             |      www.cicoria.com                |

**Comment by ines** [comment]
Thanks a lot for the fix! I wonder why it only included Python 2 before, I wish I had left a comment when I wrote / updated that code 🤦‍♀️

Btw, do you have an idea for how we could test this reliably? I guess making two tempfiles and symlinking them seems reasonable, right? It'd be nice to actually have this covered on Travis and Appveyor.

File: spacy/compat.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def getattr_(obj, name, *default):
 
 
 def symlink_to(orig, dest):
-    if is_python2 and is_windows:
+    if (is_python2 or is_python3) and is_windows:
```
Comment (ines) [inline spacy/compat.py]: In this case, I guess we could also just do the following, right?

```suggestion
    if is_windows:
```

**Comment by cicorias** [comment]
I’ll look later to add a test. Yea not sure why it was like that that’s why I didn’t leave it at just isWindows at first.

**Comment by cicorias** [comment]
@ines added a test and new function - please take a look... tx

**Comment by ines** [comment]
Nice, thanks so much! 👍 Definitely feels good having a test like this now!

**Comment by cicorias** [comment]
The logic you HAD on creating the link seems to work for Unlink. In that the Windows + Py 2.x it was treated like a Directory and not a symbolic link. Perhaps that’s where you originally had thought about why ?

**Pull request merged**